prov/sockets/src/sock_ep.c

```diff
@@ -905,16 +933,32 @@ static int sock_ep_control(struct fid *fid, int command, void *arg)
 		if (!new_ep)
 			return -FI_ENOMEM;
 		*new_ep = *ep;
-		new_ep->op_flags = alias->flags;
+		ret = sock_ep_setopflags(new_ep, alias->flags);
+		if (ret) {
+			free(new_ep);
+			return -FI_EINVAL;
+		}
 		*alias->fid = &new_ep->ep.fid;
 		break;
-
 	case FI_GETOPSFLAG:
-		*(uint64_t *) arg = ep->op_flags;
+		new_flags = *(uint64_t *) arg;
+		if ((new_flags & FI_TRANSMIT) && (new_flags & FI_RECV)) {
+			SOCK_LOG_ERROR("Both Tx/Rx flags cannot be specified\n");
+			return -FI_EINVAL;
+		} else if (new_flags & FI_TRANSMIT) {
+			*(uint64_t *) arg = ep->tx_attr.op_flags;
+		} else if (new_flags & FI_RECV) {
+			*(uint64_t *) arg = ep->rx_attr.op_flags;
+		} else {
+			SOCK_LOG_ERROR("Tx/Rx flags not specified\n");
+			return -FI_EINVAL;
+		}
```

Comment:
Do we need this same logic in sock_ctx_control?  If so, maybe separate it out similar to what you did for setopflags.